test/unit_tests/braket/experimental/autoqasm/test_decorator_gate.py

```diff
@@ -0,0 +1,271 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+"""AutoQASM tests exercising the @aq.gate decorator and related functionality.
+"""
+
+import numpy as np
+import pytest
+from test_api import _test_on_local_sim
+
+import braket.experimental.autoqasm as aq
+from braket.experimental.autoqasm import errors
+from braket.experimental.autoqasm.instructions import h, measure, rx, rz, x
+
+
+@aq.gate
```

Comment:
Fantastic tests! Really top-notch, catching lots of edge cases and interesting/creative usage paths. I feel like I understand the feature well reading these :)

Reply:
Thanks! Although I'd also credit the code coverage requirement for forcing me down some of these unnatural paths.